### PR TITLE
docs(sourceprocessor): clarify readme about pod annotations

### LINKED ISSUE #259: sourceprocessor: document `sumologic.com/exclude` and `sumologic.com/include`
Document what values are allowed/expected in `sumologic.com/exclude` and `sumologic.com/include` pod labels in order to make them work with sourceprocessor's logic.

_Original discussion in https://github.com/SumoLogic/sumologic-otel-collector/pull/257#discussion_r716610624_

### PR DESCRIPTION
**Pull request by pmalek-sumo**
Closes #259

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
what about

> overrides `exclude` processor settings

>

File: pkg/processor/sourceprocessor/README.md [hunk 1]
```diff
@@ -70,17 +70,25 @@ processors:
...
 
-- `sumologic.com/exclude` - records from this pod are dropped
-- `sumologic.com/include` - overrides `exclude` processor settings, records from this pod are not checked against those regexes
+- `sumologic.com/exclude` - records from a pod that has this annotation set to
+  `"true"` (a string) or `true` (a boolean), are dropped,
+
+  **NOTE**: this has precedence over `sumologic.com/include` if both are set at
+  the same time for one pod.
+- `sumologic.com/include` - records from a pod that has this annotation set to
+  `"true"` (a string) or `true` (a boolean) are not checked against exclusion
```
Comment (astencel-sumo) [inline pkg/processor/sourceprocessor/README.md:81]:
```suggestion
  `"true"` are not checked against exclusion
```
Kubernetes annotation values can only be string.

It's good you mention below that the OT attribute value can be either a string or a boolean :+1: 
But when talking about annotations, I think only talking about strings is justified.

File: pkg/processor/sourceprocessor/README.md [hunk 2]
```diff
@@ -70,17 +70,25 @@ processors:
...
 ## Pod annotations
 
-The following [Kubernetes annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/)
-can be used on pods:
+The following [Kubernetes annotations][k8s_annotations_doc] can be used on pods:
 
-- `sumologic.com/exclude` - records from this pod are dropped
-- `sumologic.com/include` - overrides `exclude` processor settings, records from this pod are not checked against those regexes
+- `sumologic.com/exclude` - records from a pod that has this annotation set to
+  `"true"` (a string) or `true` (a boolean), are dropped,
```
Comment (astencel-sumo) [inline pkg/processor/sourceprocessor/README.md:76]: Same here...

Comment (pmalek-sumo) [reply pkg/processor/sourceprocessor/README.md]: Please check now

**Comment by pmalek-sumo** [comment]
> what about
> 
> > overrides `exclude` processor settings
> 
> >

Is this unclear? Should I add something to this to be more clear?

**Comment by sumo-drosiek** [comment]
It was, before latest changes

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**